Add vscode info (#107)
pingcap/tidb-dev-guide: src/get-started/setup-an-ide.md

# Setup an IDE

> Using an IDE is recommended as it makes it a lot easier to for example see the fields of a struct. However it is not required to use a specic IDE or editor.

Reviewer: what is "it is not required to use a specic IDE or editor"? even change "specic" to "specific" it still confuses me.

Author: ```suggestion
Using an IDE is recommended as it makes it a lot easier to work with the TiDB code, for example to see the fields of a struct. However it is not required to use a specific IDE or editor. You can use the IDE or editor of your choice.
```

Reviewer: Thanks for your update. I get the point.

![unit test output](../img/unit-test-output.png)

If you encounter any problems during your journey, do not hesitate to reach out on the [TiDB Internals forum](https://internals.tidb.io/).

## Visual Studio Code

VS Code is a generic IDE that has good extensions for working with Go and TiDB.

![VS Code with TiDE](../img/vscode_tide.png)

### Prerequisites

* `go`: TiDB is a Go project thus its building requires a working `go` environment. See the previous [Install Golang](install-golang.md) section to prepare the environment.
* TiDB source code: See the previous [Get the code, build and run](build-tidb-from-source.md) section to get the source code.

### Download VS Code

> Download GoLand from [here](https://code.visualstudio.com/Download) and install it.

Reviewer 2: ```suggestion
Download VS Code from [here](https://code.visualstudio.com/Download) and install it.
```

Now install these extensions:

* [Go](https://marketplace.visualstudio.com/items?itemName=golang.Go)

> * [TiDE](https://marketplace.visualstudio.com/items?itemName=dragonly.ticode)

Reviewer 2: And I think TiDE is also not mandatory? 

Also, GoLand instructions above mainly focus on simple tasks like unistore and unit tests. But TiDE is targeted at more complex tasks (and doesn't support unistore)?

Author: Yes not mandatory, but quite useful. I think users would be smart enough to filter this out if needed and if they don't that's also not really a problem.

> * [GitHub Pull Requests and Issues](https://marketplace.visualstudio.com/items?itemName=github.vscode-pull-request-github)

Reviewer 2: ```suggestion
```
I think this extension is irrelevant?

Author: I think having this extension makes sense as GitHub is core the the contribution workflow. We could mark it as optional, but then we should probably label TiDE as optional as well.
